metadata-ingestion/src/datahub/testing/compare_metadata_json.py

```diff
@@ -0,0 +1,109 @@
+"""Utilities for comparing MCE and MCP files."""
+
+import json
+import logging
+import os
+import pathlib
+import pprint
+import shutil
+import tempfile
+from typing import Any, Dict, List, Sequence, Union
+
+import pytest
+from deepdiff import DeepDiff
+
+from datahub.ingestion.sink.file import write_metadata_file
+from datahub.ingestion.source.file import read_metadata_file
+from datahub.testing.mcp_diff import MCPDiff, get_aspects_by_urn
+
+logger = logging.getLogger(__name__)
+
+MetadataJson = List[Dict[str, Any]]
+
+default_exclude_paths = [
+    r"root\[\d+]\['systemMetadata']\['lastObserved']",
+    r"root\[\d+]\['aspect']\['json']\['timestampMillis']",
+    r"root\[\d+]\['aspect']\['json']\['lastUpdatedTimestamp']",
+    r"root\[\d+]\['aspect']\['json']\['created']",
+    r"root\[\d+]\['aspect']\['json']\['lastModified']",
+]
+
+
+def load_json_file(filename: Union[str, os.PathLike]) -> MetadataJson:
+    with open(str(filename)) as f:
+        return json.load(f)
+
+
+def assert_metadata_files_equal(
+    output_path: Union[str, os.PathLike],
+    golden_path: Union[str, os.PathLike],
+    update_golden: bool,
+    copy_output: bool,
+    ignore_paths: Sequence[str] = (),
+) -> None:
+    golden_exists = os.path.isfile(golden_path)
+
+    if copy_output:
+        shutil.copyfile(str(output_path), str(golden_path) + ".output")
+        print(f"Copied output file to {golden_path}.output")
+
+    if not update_golden and not golden_exists:
+        raise FileNotFoundError(
+            "Golden file does not exist. Please run with the --update-golden-files option to create."
+        )
+
+    output = load_json_file(output_path)
+
+    # if updating a golden file that doesn't exist yet, load the output again
+    if update_golden and not golden_exists:
+        golden = load_json_file(output_path)
+        shutil.copyfile(str(output_path), str(golden_path))
```

Comment:
shouldn't we just copy the file and return?

why do we even fall through and run the diff logic if `update_golden` is true?

Reply:
Good point